tests/next_tests/integration_tests/feature_tests/ffront_tests/test_compiled_program.py

```diff
@@ -436,6 +436,62 @@ def test_compile_variants_not_compiled(cartesian_case, compile_variants_testee):
         )
 
 
+def test_compile_variants_not_compiled_but_jit_enabled_on_call(
+    cartesian_case, compile_variants_testee
+):
+    # disable jit on the program
+    object.__setattr__(compile_variants_testee, "enable_jit", False)
+
+    field_a = cases.allocate(cartesian_case, compile_variants_testee, "field_a")()
+    field_b = cases.allocate(cartesian_case, compile_variants_testee, "field_b")()
+    out = cases.allocate(cartesian_case, compile_variants_testee, "out")()
+
+    compile_variants_testee(
+        field_a,
+        int32(3),  # variant does not exist
+        4.0,
+        False,
+        field_b,
+        out=out,
+        offset_provider=cartesian_case.offset_provider,
+        enable_jit=True,  # but enable jit on the call
+    )
+    assert np.allclose(out[0].ndarray, field_a.ndarray - 3)
+    assert np.allclose(out[1].ndarray, field_b.ndarray - 4.0)
+
+
+def test_compile_variants_config_default_disable_jit(cartesian_case, compile_variants_testee):
+    field_a = cases.allocate(cartesian_case, compile_variants_testee, "field_a")()
+    field_b = cases.allocate(cartesian_case, compile_variants_testee, "field_b")()
+    out = cases.allocate(cartesian_case, compile_variants_testee, "out")()
+
+    # the default in config is to allow jit (if that changes, this test has to change)
```

Comment:
Why not just patching the default to make sure is True as it is done in the False case?

Reply:
Makes sense. Since the test is about changing the `config` default before call, I thought I have to rely on the current value, but obviously one of the 2 will always be the non-default.